core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.html

```diff
@@ -1,12 +1,12 @@
 <div role="main" class="oppia-conversation-skin-animate-cards-container position-relative" *ngIf="hasFullyLoaded">
   <div class="conversation-skin-cards-container conversation-skin-animate-cards e2e-test-conversation-skin-cards-container"
-       [ngClass]="{'animate-to-two-cards': isAnimatingToTwoCards, 'animate-to-one-card': isAnimatingToOneCard}">
+       [ngClass]="{'animate-to-two-cards': conversationSkinService.isPlayerAnimatingToOneCard(), 'animate-to-one-card': conversationSkinService.isPlayerAnimatingToTwoCards()}">
```

Comment:
In general the template shouldn't call the service directly. You can have pass-through functions in the controller that redirect to the service function.

That said, I feel like "is the player currently animating" is more of a UI thing and you probably want to track that specific detail in the controller. (If you think about it -- it's one of the things that would typically change if you reskinned the learner UI.) Use the service more for state-based operations, e.g. maintaining the stack of cards in the learner UI accurately.

Reply:
That makes sense but we need the `animateToOneCard` and `animateToTwoCards` functions in the service - and they are responsible for handling the state of the "isPlayerAnimating..." variables, which is why I moved them here.

We don't really need any such animations currently for the new skin but since they are right in the middle of common code, it's simply very convenient to just include them. Otherwise, we will have to duplicate the main conditions that change the displayed card. What do you suggest?

Reply:
IIUC you misinterpreted Sean's comment. Just add proxies, for every function from the service that you call from html add a corresponding function in the conversation-skin.component.

Like
```
isPlayerAnimatingToOneCard(): boolean {
    return service.isPlayerAnimatingToOneCard();
  }
```

Reply:
Ah okay, got it.